runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -436,6 +427,27 @@
         @Default
         List<MsgHandlingFilter> nettyClientInboundMsgFilters = null;
 
+        /**
+         * The default priority of the requests made by this client.
+         * Under resource constraints non-high priority requests
+         * are dropped.
+         */
+        @Default
+        PriorityLevel priorityLevel = PriorityLevel.NORMAL;
+
+        /**
+         * Port at which the {@link CorfuRuntime} will allow third-party
+         * collectors to pull for metrics.
+         */
+        @Default
+        int prometheusMetricsPort = MetricsUtils.NO_METRICS_PORT;
+
+        /**
+         * The compression codec to use to encode a write's payload
+         */
+        @Default
+        String codecType = Codec.Type.ZSTD.toString();
```

Comment:
Also, probably should be final

Reply:
I had asked Maithem this, and this was his answer:

This will be exposed to the applications, a string is more flexible than our Codec.Type, just in case we need to refactor, I don't want to break applications code.